coordinator/gscoordinator/coordinator.py

```diff
@@ -379,13 +393,29 @@ def _generate_runstep_request(session_id, dag_def, dag_bodies):
         # preprocess of op before run on analytical engine
         for op in dag_def.op:
             self._key_to_op[op.key] = op
-            op_pre_process(
-                op,
-                self._op_result_pool,
-                self._key_to_op,
-                engine_hosts=self._engine_hosts,
-                engine_config=self._analytical_engine_config,
-            )
+            if (
+                op.op == types_pb2.CREATE_GRAPH or op.op == types_pb2.RUN_APP
+            ) and self._analytical_engine_config["enable_java_sdk"] == "ON":
+                logger.info(
+                    "op preprocess for java sdk enabled, and op is one of create graph and run app"
+                )
+                op_pre_process(
+                    op,
+                    self._op_result_pool,
+                    self._key_to_op,
+                    engine_hosts=self._engine_hosts,
+                    engine_config=self._analytical_engine_config,
+                    engine_java_class_path=self._java_class_path,
+                    engine_jvm_opts=self._jvm_opts,
+                )
+            else:
+                op_pre_process(
+                    op,
+                    self._op_result_pool,
+                    self._key_to_op,
+                    engine_hosts=self._engine_hosts,
+                    engine_config=self._analytical_engine_config,
+                )
```

Comment:
The `if-else` here is not needed, right?

The code is `else` branch looks equivalent to the `if` branch and the last two parameters should just be `None`.

Reply:
+1

Reply:
if-else removed